libsplinter/src/biome/credentials/rest_api/actix_web_1/mod.rs

```diff
@@ -0,0 +1,266 @@
+                self.credentials_config.clone(),
+                self.credentials_store.clone(),
+                self.key_store.clone(),
+            ));
+        }
+
+        resources
+    }
+}
+
+/// Builder for BiomeCredentialsRestResourceProvider
+#[derive(Default)]
+pub struct BiomeCredentialsRestResourceProviderBuilder {
+    #[cfg(feature = "biome-key-management")]
+    key_store: Option<Arc<dyn KeyStore>>,
+    credentials_config: Option<BiomeCredentialsRestConfig>,
+    token_secret_manager: Option<Arc<dyn SecretManager>>,
+    refresh_token_secret_manager: Option<Arc<dyn SecretManager>>,
+    refresh_token_store: Option<Arc<dyn RefreshTokenStore>>,
+    credentials_store: Option<Arc<dyn CredentialsStore>>,
+}
+
+impl BiomeCredentialsRestResourceProviderBuilder {
+    /// Sets a KeyStore for the BiomeCredentialsRestResourceProvider
+    ///
+    /// # Arguments
+    ///
+    /// * `pool`: ConnectionPool to database that will serve as backend for KeyStore
+    #[cfg(feature = "biome-key-management")]
+    pub fn with_key_store(
+        mut self,
+        store: impl KeyStore + 'static,
+    ) -> BiomeCredentialsRestResourceProviderBuilder {
+        self.key_store = Some(Arc::new(store));
+        self
+    }
+
+    /// Sets a BiomeCredentialsRestConfig for the BiomeCredentialsRestResourceProvider
+    ///
+    /// # Arguments
+    ///
+    /// * `config`: the BiomeCredentialsRestConfig that will be used to configure the Biome resources
+    pub fn with_credentials_config(
+        mut self,
+        config: BiomeCredentialsRestConfig,
+    ) -> BiomeCredentialsRestResourceProviderBuilder {
+        self.credentials_config = Some(config);
+        self
+    }
+
+    /// Sets a CredentialsStore for the BiomeCredentialsRestResourceProvider
+    ///
+    /// # Arguments
+    ///
+    /// * `pool`: ConnectionPool to database that will serve as backend for CredentialsStore
+    pub fn with_credentials_store(
+        mut self,
+        store: impl CredentialsStore + 'static,
+    ) -> BiomeCredentialsRestResourceProviderBuilder {
+        self.credentials_store = Some(Arc::new(store));
+        self
+    }
+
+    /// Sets a SecretManager for JWT tokens for the BiomeCredentialsRestResourceProvider
+    ///
+    /// # Arguments
+    ///
+    /// * `secret_manager`: the SecretManager to be used for fetching and generating secrets to
+    ///   sign and verify JWT tokens
+    pub fn with_token_secret_manager(
+        mut self,
+        secret_manager: impl SecretManager + 'static,
+    ) -> BiomeCredentialsRestResourceProviderBuilder {
+        self.token_secret_manager = Some(Arc::new(secret_manager));
+        self
+    }
+
+    /// Sets a SecretManager for the refresh tokens for the BiomeCredentialsRestResourceProvider
+    ///
+    /// # Arguments
+    ///
+    /// * `secret_manager`: the SecretManager to be used for fetching and generating secrets to
+    ///   sign and verify JWT tokens
```

Comment:
Comment here looks like copypasta from the above function

Reply:
Fixed